client/me/account/main.jsx

```diff
@@ -492,7 +502,8 @@ const Account = React.createClass( {
 						id="url"
 						name="url"
 						onFocus={ this.recordFocusEvent( 'Web Address Field' ) }
-						valueLink={ this.valueLink( 'user_URL' ) }
+						value={ this.getUserSetting( 'user_URL' ) || '' }
+						onChange={ this.updateUserSettingInput( 'user_URL' ) }
```

Comment:
I think it's also very useful if the linter doesn't let you put too much complex code into a JSX attribute, and forces you to extract the logic to a separate method. Even if there is no performance benefit from that.

But anyway, putting the `settingName` to the JSX markup is a great idea! We don't even need additional `data` attributes, as the name is already there in the `id` and `name` properties.

I updated the `id`, `name` and `htmlFor` attribute values to be consistent with the `userSettings` property names, and rewrote the `updateUserSetting*` methods to get the setting name from `event.target.name`.